lib/blocktree/blocktree_test.go

```diff
@@ -371,7 +371,141 @@ func TestBlockTree_IsDecendantOf(t *testing.T) {
 	require.False(t, isDescendant)
 }
 
-func TestBlockTree_HighestCommonAncestor(t *testing.T) {
+func Test_Node_LowestCommonAncestorNew(t *testing.T) {
```

Comment:
nit we test `lowestCommonAncestor` so it should be:

```suggestion
func Test_lowestCommonAncestor(t *testing.T) {
```